### PR TITLE
Fix #9579 - ignore disabled input in Abide

### LINKED ISSUE #9579: [Abide] Disabled fields not being ignored in validation
**How can we reproduce this bug?**

1. Use the example abide form code from the documentation
2. Add a disabled attribute to the first required field
3. Submit the form for checking

**What did you expect to happen?**
I expected Abode to ignore the disabled field as it does in  in Foundation 5
**What happened instead?**
The disabled field is highlighted with an error
**Test case:**
http://codepen.io/colinmackinlay/pen/PWYZGw

### PR DESCRIPTION
**Pull request by colinmackinlay**
@ncoden To resolve #9579 so that disabled inputs are not validated

### CODE REVIEW

**Review by ncoden — APPROVED** [review]

**Comment by ncoden** [comment]
Next time, please use a better name for commit, like `Fix #9579 - ignore disabled input in Abide`

**Comment by colinmackinlay** [comment]
Sorry about that - realised after I'd moved longer description to the body.

Tried to update docs but can't see how to edit "Javascript Reference" which seems the most appropriate place. The abide.md file doesn't include this or the SASS reference. Presumably this is generated elsewhere?

**Comment by ncoden** [comment]
"Javascript Reference" is generated from the comments above functions and options, like [here](https://github.com/ncoden/foundation-sites/blob/a87eb5dc02f714f80bd3648151d2c53185827548/js/foundation.abide.js#L84). But I think we should also add a section to the docs (like `Ignored inputs` with an example for `[data-abide-ignore]`, `[type="hidden"]` and [disabled]`).

File: docs/pages/abide.md [hunk 1]
```diff
@@ -34,6 +34,18 @@ These input types create a text field: `text`, `date`, `datetime`, `datetime-loc
       <p class="help-text" id="exampleHelpTex">This input is ignored by Abide using `data-abide-ignore`</p>
     </div>
     <div class="small-12 columns">
+      <label>Disabled!
+        <input type="text" placeholder="You can't use me!" aria-describedby="exampleHelpTex" disabled>
```
Comment (ncoden) [inline docs/pages/abide.md]: `placeholder="Disabled input"`

File: docs/pages/abide.md [hunk 2]
```diff
@@ -34,6 +34,18 @@ These input types create a text field: `text`, `date`, `datetime`, `datetime-loc
...
+      </label>
+      <p class="help-text" id="exampleHelpTex">This input is ignored by Abide using `disabled`</p>
+    </div>
+    <div class="small-12 columns">
+      <label>Hidden!
+        <input type="hidden" placeholder="You can't see me!" aria-describedby="exampleHelpTex" >
```
Comment (ncoden) [inline docs/pages/abide.md]: placeholder="Hidden input"

**Comment by colinmackinlay** [comment]
validateInput() comment updated and I've added two further examples below the data-abide-ignore one.
Perhaps the [type="hidden"] one could be clearer but it's hard to see how!

File: docs/pages/abide.md (hunk 1 above)
Comment (ncoden) [inline docs/pages/abide.md]: Also, don't you think there is already too much inputs in this example, and we should rather move ignored input to their own section ?

**Comment by colinmackinlay** [comment]
Section before Initial State seems to just hang without explanation. It seems to be an example of how to show error state initially both in the label and with callout. Is that right @gakimball ?

**Review by ncoden — APPROVED** [review]

**Comment by kball** [comment]
Merged!  Thanks @colinmackinlay

**Pull request merged**